File: lib/RedeNeural/Arquivo.php
<?php
namespace RedeNeural;

class Arquivo{
    
    private $caminhoPasta;
    
    public function __construct(){
        $this->caminhoPasta = dirname(dirname(dirname(__FILE__)));
    }
    
    private function buscarConteudoArquivo($nomeArquivo){
        $conteudo = '';
                
        //Abre arquivo
        $arquivo = fopen ($this->caminhoPasta.$nomeArquivo, "r");
        
        if($arquivo){
            //Read file, line to line
            while (!feof ($arquivo)) {
                //Read one line
                $conteudo.= trim(fgets($arquivo, 4096));        
            }
            //Close$arquivo
            fclose ($arquivo);
        }
        
        return $conteudo;
    }
    
    private function scanear($pastaSelecionada){
        
        $pasta = $this->caminhoPasta.$pastaSelecionada;
        
        $arquivos = scandir($pasta);
        return array_diff($arquivos, array('..', '.'));
    }
    
    public function buscarArquivosTreinamento(){
        
        $pasta = '/treino/';
        $arquivos = $this->scanear($pasta);
        
        
        $entradas = array();
        
        foreach($arquivos as $arquivo){
            $entradas[] = $this->buscarConteudoArquivo($pasta.$arquivo);
        }
        
        return $entradas;
        
    }
    
    public function buscarArquivosClassificacao(){
        
        $pasta = '/classificacao/';
        
        $arquivos = $this->scanear($pasta);
        
        $entradas = array();
                
        foreach($arquivos as $arquivo){
            $entradas[] = $this->buscarConteudoArquivo($pasta.$arquivo);
        }
        
        return $entradas;
    }
        
}

File: lua/Arquivo.lua
RedeNeural.Arquivo = {};


function RedeNeural.Arquivo:pastaArquivo()
	return ''
end

function RedeNeural.Arquivo:_buscarConteudoArquivo(nomeArquivo)

	local conteudo = ''
	local file = io.open(nomeArquivo)
	
	if file then
		conteudo = file:read('*a')
	end

	return conteudo

end

function RedeNeural.Arquivo:_scanear(pastaSeleciona)
	local files = {}
	local i = 0

	local p = io.popen('find "'..pastaSeleciona..'" -type f')
	for file in p:lines() do
		files[i] = file
		i = i + 1
	end

	return files;

end

function RedeNeural.Arquivo:_buscarArquivos(nomePasta)
	local entradas = {}
	local arquivos = self:_scanear(nomePasta)
	for pos, val in pairs(arquivos) do
		entradas[pos] = self:_buscarConteudoArquivo(val)
	end
	return entradas
end

function RedeNeural.Arquivo:buscarArquivosTreinamento()
	return self:_buscarArquivos('treino')
end

function RedeNeural.Arquivo:buscarArquivosClassificacao()
	return self:_buscarArquivos('classificacao')
end